### PR TITLE
Mentioned NFT-Preview tool

### PR DESCRIPTION
**Pull request by Utkarshbhimte**
I built this tool to make it easy for me to get a quick preview of the `tokenURI` content and the image.

### CODE REVIEW

**Comment by Utkarshbhimte** [comment]
Hey @adilanchian, sure. I will add some more content.

Not sure how it can be important, it's just there to make it easier to experiment with different designs and debug randomly generated text content and also a quick check if the `tokenURI` data is valid or not (all those manual string concatenation are bug-prone and painful to debug)

**Comment by adilanchian** [comment]
Ya sorry I should have been a bit more specific about this - 

This section just feels a bit out of place. Like why do I need to debug my `tokenURI`? So something along the lines of:

"Now that you have your `tokenURI` setup, how do we know if it's actually correct? After all, this holds all our data for our NFT! You can use a cool tool like - {insert tool} to see exactly what data this thing is holding."

You see what I mean? Let me know and I can write it up if you want :)

**Comment by Utkarshbhimte** [comment]
Thanks for that @adilanchian, I have made some changes there. 

Writing is not my strong suit but I appreciate all the help. 😊

File: NFT_Collection/en/Section_2/Lesson_3_Dynamically_Generating_NFTs.md [hunk 1]
```diff
@@ -159,6 +159,30 @@ abi.encodePacked("data:application/json;base64,", json)
...
+        string(
+            abi.encodePacked(
+                "https://nftpreview.0xdev.codes/?code=",
+                finalTokenUri
+            )
+        )
+    );
+    console.log("--------------------\n");
+```
+![](https://imgur.com/a/cfZiWh8)
```
Comment (adilanchian) [inline NFT_Collection/en/Section_2/Lesson_3_Dynamically_Generating_NFTs.md:184]: Hmm I don't see an image preview here when looking at the non-raw version of the markdown! Can you verify this for me?

Thanks!

**Review by adilanchian — CHANGES_REQUESTED** [review]
Yessss looks great!

I see that you added an image here, but it doesn't seem to render for me? Can you verify that this is looking okay?

**Comment by Utkarshbhimte** [comment]
Will check and share soon [image: 😊]

Regards
Utkarsh Bhimte

**Comment by adilanchian** [comment]
Awesome :)

Ya just for clarity this it what GitHub is showing me:

image


^ looking at this I think you for got to add `[image]` in the bracket!

File: NFT_Collection/en/Section_2/Lesson_3_Dynamically_Generating_NFTs.md (hunk 1 above)
Comment (adilanchian) [reply NFT_Collection/en/Section_2/Lesson_3_Dynamically_Generating_NFTs.md:184]: Looks like here you just need to put: 

`image`

Comment (Utkarshbhimte) [reply NFT_Collection/en/Section_2/Lesson_3_Dynamically_Generating_NFTs.md]: done

File: NFT_Collection/en/Section_2/Lesson_3_Dynamically_Generating_NFTs.md [hunk 2]
```diff
@@ -159,6 +159,30 @@ abi.encodePacked("data:application/json;base64,", json)
...
+        string(
+            abi.encodePacked(
+                "https://nftpreview.0xdev.codes/?code=",
+                finalTokenUri
+            )
+        )
+    );
+    console.log("--------------------\n");
+```
+![image](https://imgur.com/a/cfZiWh8)
```
Comment (adilanchian) [inline NFT_Collection/en/Section_2/Lesson_3_Dynamically_Generating_NFTs.md:184]: Hehe this is still slightly broken :)

You need to get the Image Address which is actually - `https://i.imgur.com/CsBxROj.png`

**Review by adilanchian — CHANGES_REQUESTED** [review]

File: NFT_Collection/en/Section_2/Lesson_3_Dynamically_Generating_NFTs.md [hunk 3]
```diff
@@ -159,6 +159,30 @@ abi.encodePacked("data:application/json;base64,", json)
...
+    console.log(
+        string(
+            abi.encodePacked(
+                "https://nftpreview.0xdev.codes/?code=",
+                finalTokenUri
+            )
+        )
+    );
+    console.log("--------------------\n");
+```
```
Comment (adilanchian) [inline NFT_Collection/en/Section_2/Lesson_3_Dynamically_Generating_NFTs.md:183]: Hey sorry one more thing here - 

Looks like you have an extra tab on every line here can you remove those please :)?

**Review by adilanchian — CHANGES_REQUESTED** [review]

**Comment by Utkarshbhimte** [comment]
Done 😊

**Comment by adilanchian** [comment]
WOWZA. It's time. Very nice work and thanks for the contribution!

**Comment by adilanchian** [comment]
@all-contributors please add @Utkarshbhimte  for doc

**Comment by Utkarshbhimte** [comment]
Thanks @adilanchian , I am planning to make it better and add all the opensea attributes here.
Make it a full-fletched opensea preview. 😊

**Pull request merged**